# WTA Data from 2000 to 2016

https://www.kaggle.com/gmadevs/wta-matches


In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('wta_matches_2000.csv')
print("Year: 2000 \t Dataframe shape: ", df.shape)

df = pd.read_csv('wta_matches_2001.csv')
print("Year: 2001 \t Dataframe shape: ", df.shape)

df = pd.read_csv('wta_matches_2002.csv')
print("Year: 2002 \t Dataframe shape: ", df.shape)

df = pd.read_csv('wta_matches_2003.csv')
print("Year: 2003 \t Dataframe shape: ", df.shape)

df = pd.read_csv('wta_matches_2004.csv')
print("Year: 2004 \t Dataframe shape: ", df.shape)

df = pd.read_csv('wta_matches_2005.csv')
print("Year: 2005 \t Dataframe shape: ", df.shape)

df = pd.read_csv('wta_matches_2006.csv')
print("Year: 2006 \t Dataframe shape: ", df.shape)

df = pd.read_csv('wta_matches_2007.csv')
print("Year: 2007 \t Dataframe shape: ", df.shape)

df = pd.read_csv('wta_matches_2008.csv')
print("Year: 2008 \t Dataframe shape: ", df.shape)

df = pd.read_csv('wta_matches_2009.csv')
print("Year: 2009 \t Dataframe shape: ", df.shape)

df = pd.read_csv('wta_matches_2010.csv')
print("Year: 2010 \t Dataframe shape: ", df.shape)

df = pd.read_csv('wta_matches_2011.csv')
print("Year: 2011 \t Dataframe shape: ", df.shape)

df = pd.read_csv('wta_matches_2012.csv', encoding = "ISO-8859-1")
print("Year: 2012 \t Dataframe shape: ", df.shape)

df = pd.read_csv('wta_matches_2013.csv')
print("Year: 2013 \t Dataframe shape: ", df.shape)

df = pd.read_csv('wta_matches_2014.csv')
print("Year: 2014 \t Dataframe shape: ", df.shape)

df = pd.read_csv('wta_matches_2015.csv')
print("Year: 2015 \t Dataframe shape: ", df.shape)

df = pd.read_csv('wta_matches_2016.csv')
print("Year: 2016 \t Dataframe shape: ", df.shape)

Year: 2000 	 Dataframe shape:  (2893, 30)
Year: 2001 	 Dataframe shape:  (3098, 30)
Year: 2002 	 Dataframe shape:  (3140, 30)
Year: 2003 	 Dataframe shape:  (2933, 49)
Year: 2004 	 Dataframe shape:  (2805, 49)
Year: 2005 	 Dataframe shape:  (2843, 49)
Year: 2006 	 Dataframe shape:  (2787, 49)
Year: 2007 	 Dataframe shape:  (2778, 49)
Year: 2008 	 Dataframe shape:  (2790, 49)
Year: 2009 	 Dataframe shape:  (2722, 49)
Year: 2010 	 Dataframe shape:  (2781, 49)
Year: 2011 	 Dataframe shape:  (2804, 49)
Year: 2012 	 Dataframe shape:  (2910, 49)
Year: 2013 	 Dataframe shape:  (2776, 49)
Year: 2014 	 Dataframe shape:  (2785, 49)
Year: 2015 	 Dataframe shape:  (2651, 49)
Year: 2016 	 Dataframe shape:  (91, 49)


# Sparsity statistics from 2003-2016

Since the number of attrtibutes have increased from 30 to 49 from the year 2003, I am concatinating the data for the years 2003-2016 for further analysis.

In [3]:
df2003 = pd.read_csv('wta_matches_2003.csv')
df2004 = pd.read_csv('wta_matches_2004.csv')
df2005 = pd.read_csv('wta_matches_2005.csv')
df2006 = pd.read_csv('wta_matches_2006.csv')
df2007 = pd.read_csv('wta_matches_2007.csv')
df2008 = pd.read_csv('wta_matches_2008.csv')
df2009 = pd.read_csv('wta_matches_2009.csv')
df2010 = pd.read_csv('wta_matches_2010.csv')
df2011 = pd.read_csv('wta_matches_2011.csv')
df2012 = pd.read_csv('wta_matches_2012.csv', encoding = "ISO-8859-1")
df2013 = pd.read_csv('wta_matches_2013.csv')
df2014 = pd.read_csv('wta_matches_2014.csv')
df2015 = pd.read_csv('wta_matches_2015.csv')
df2016 = pd.read_csv('wta_matches_2016.csv')

frames = [df2003, df2004, df2005, df2006, df2007, df2008, df2009, df2010, df2011, df2012, df2013, df2014, df2015, df2016] 
df = pd.concat(frames, sort = 'False')
display(df)

best_of  draw_size  l_1stIn  l_1stWon  l_2ndWon  l_SvGms  l_ace  \
0         3        128     35.0      14.0       4.0      NaN    0.0   
1         3        128     88.0      50.0      11.0      NaN    3.0   
2         3        128     59.0      33.0      14.0      NaN    2.0   
3         3        128     33.0      17.0      10.0      NaN    0.0   
4         3        128     69.0      47.0      26.0      NaN    4.0   
5         3        128     38.0      26.0       8.0      NaN    4.0   
6         3        128     32.0      16.0      11.0      NaN    2.0   
7         3        128     82.0      47.0      10.0      NaN    0.0   
8         3        128     32.0      19.0      10.0      NaN    1.0   
9         3        128     49.0      19.0       8.0      NaN    0.0   
10        3        128     40.0      18.0      14.0      NaN    0.0   
11        3        128     30.0      15.0       2.0      NaN    0.0   
12        3        128     38.0      24.0       8.0      NaN    1.0   
13        3        128     21.0       7.0       6.0      NaN    0.0   
14        3        128     40.0      17.0       5.0      NaN    0.0   
15        3        128     45.0      25.0      11.0      NaN    1.0   
16        3        128     36.0      18.0      16.0      NaN    0.0   
17        3        128     54.0      37.0      22.0      NaN    6.0   
18        3        128     44.0      24.0       8.0      NaN    1.0   
19        3        128     34.0      20.0      10.0      NaN    2.0   
20        3        128     27.0      14.0       6.0      NaN    0.0   
21        3        128     30.0      17.0       7.0      NaN    2.0   
22        3        128     42.0      21.0      10.0      NaN    0.0   
23        3        128     32.0      19.0      10.0      NaN    1.0   
24        3        128     29.0      16.0       6.0      NaN    0.0   
25        3        128     25.0      16.0       3.0      NaN    0.0   
26        3        128     31.0      18.0       8.0      NaN    0.0   
27        3        128     38.0      20.0       9.0      NaN    1.0   
28        3        128     43.0      23.0      11.0      NaN    0.0   
29        3        128     53.0      30.0      12.0      NaN    2.0   
..      ...        ...      ...       ...       ...      ...    ...   
61        3         32     36.0      23.0       7.0      9.0    1.0   
62        3         32     39.0      22.0      10.0      9.0    6.0   
63        3         32     31.0      20.0       9.0      8.0    3.0   
64        3         32     58.0      42.0      12.0     13.0    3.0   
65        3         32     38.0      26.0      15.0     10.0    6.0   
66        3         32     51.0      34.0      12.0     12.0    7.0   
67        3         32     40.0      19.0      10.0     10.0    2.0   
68        3         32     48.0      32.0      14.0     10.0    0.0   
69        3         32     35.0      25.0      12.0      8.0    4.0   
70        3         32     31.0      16.0       4.0      8.0    2.0   
71        3         32     33.0      20.0       7.0      9.0    1.0   
72        3         32     36.0      19.0       3.0      8.0    0.0   
73        3         32     58.0      34.0      16.0     14.0    2.0   
74        3         32     30.0      13.0       7.0      8.0    0.0   
75        3         32     31.0      24.0       9.0      9.0    3.0   
76        3         32     29.0      16.0      11.0      9.0    0.0   
77        3         32     46.0      27.0      11.0     11.0    2.0   
78        3         32     45.0      32.0      12.0     13.0    3.0   
79        3         32     31.0      15.0       9.0      8.0    0.0   
80        3         32     24.0      15.0       6.0      7.0    3.0   
81        3         32     31.0      17.0      14.0      9.0    3.0   
82        3         32     48.0      28.0      11.0     11.0    0.0   
83        3         32     53.0      32.0      17.0     14.0    3.0   
84        3         32     30.0      17.0       4.0      7.0    1.0   
85        3         32     3

In [4]:
df.shape

(36456, 49)

In [5]:
df.isna().mean()*100

best_of                0.000000
draw_size              0.000000
l_1stIn               45.775730
l_1stWon              45.775730
l_2ndWon              45.775730
l_SvGms               99.717468
l_ace                 45.794931
l_bpFaced             45.781216
l_bpSaved             45.781216
l_df                  45.874479
l_svpt                45.778473
loser_age              0.010972
loser_entry           79.361422
loser_hand             1.730854
loser_ht              38.646588
loser_id               0.000000
loser_ioc              0.000000
loser_name             0.000000
loser_rank             4.616524
loser_rank_points      4.616524
loser_seed            77.073733
match_num              0.000000
minutes               99.717468
round                  0.000000
score                  0.000000
surface                0.117950
tourney_date           0.000000
tourney_id             0.000000
tourney_level          0.000000
tourney_name           0.000000
w_1stIn               45.775730
w_1stWon

#  Sparsity Statistics from 2000 to 2002

In [6]:
df2000 = pd.read_csv('wta_matches_2000.csv')
df2001 = pd.read_csv('wta_matches_2001.csv')
df2002 = pd.read_csv('wta_matches_2002.csv')

frames = [df2000, df2001, df2002]
df = pd.concat(frames, sort = 'False')
display(df)

best_of  draw_size  loser_age loser_entry loser_hand  loser_ht  \
0           3        128  17.859001         NaN          R       NaN   
1           3        128  27.118412           Q          R       NaN   
2           3        128  31.378508         NaN          R       NaN   
3           3        128  22.006845         NaN          R       NaN   
4           3        128  24.821355         NaN          R       NaN   
5           3        128  18.850103         NaN          R     180.0   
6           3        128  26.746064           Q          R       NaN   
7           3        128  21.998631         NaN          R       NaN   
8           3        128  23.657769         NaN          R       NaN   
9           3        128  26.151951           Q          R       NaN   
10          3        128  29.708419         NaN          R       NaN   
11          3        128  25.650924         NaN          R       NaN   
12          3        128  20.257358         NaN          R       NaN   
13          3        128  28.826831         NaN          R       NaN   
14          3        128  25.218344          WC          R       NaN   
15          3        128  21.998631         NaN          R       NaN   
16          3        128  20.815880          WC          L       NaN   
17          3        128  22.193018         NaN          R       NaN   
18          3        128  21.954825         NaN          R       NaN   
19          3        128  23.504449         NaN          R       NaN   
20          3        128  27.326489         NaN          R       NaN   
21          3        128  16.684463          WC          R     173.0   
22          3        128  22.644764         NaN          R       NaN   
23          3        128  25.511294         NaN          R       NaN   
24          3        128  21.694730         NaN          R       NaN   
25          3        128  25.812457         NaN          R       NaN   
26          3        128  24.865161         NaN          R       NaN   
27          3        128  23.852156         NaN          R       NaN   
28          3        128  25.404517         NaN          R       NaN   
29          3        128  22.548939           Q          R     172.0   
...       ...        ...        ...         ...        ...       ...   
3110        3          4  28.098563         NaN          R       NaN   
3111        3          4  20.158795         NaN          R       NaN   
3112        3          4  24.490075         NaN          R       NaN   
3113        3          4  21.273101         NaN          R       NaN   
3114        3          4  24.490075         NaN          R       NaN   
3115        3          4  21.273101         NaN          R       NaN   
3116        3          4  19.581109         NaN          R       NaN   
3117        3          4  18.735113         NaN          R     174.0   
3118        3          4  19.581109         NaN          R       NaN   
3119        3          4  18.735113         NaN          R     174.0   
3120        3          4  23.583847         NaN          L     168.0   
3121        3          4  17.067762         NaN          R       NaN   
3122        3          4  27.080082         NaN          R       NaN   
3123        3          4  18.409309         NaN          R     176.0   
3124        3          4  25.995893         NaN          R       NaN   
3125        3          4  29.237509         NaN          R       NaN   
3126        3          4  25.995893         NaN          R       NaN   
3127        3          4  29.237509         NaN          R       NaN   
3128        3          4  24.229979         NaN          R       NaN   
3129        3          4  26.633812         NaN          R     176.0   
3130        3          4  30.532512         NaN          R     170.0   
3131        3          4  30.861054         NaN          R       NaN   
3132        3          4  19.515400         NaN          R     181.0   
3133        3          4  30.502396         Na

In [7]:
df.shape

(9131, 30)

In [8]:
df.isna().mean()*100

best_of                0.000000
draw_size              0.000000
loser_age              0.000000
loser_entry           80.144562
loser_hand             4.008323
loser_ht              73.967802
loser_id               0.000000
loser_ioc              0.000000
loser_name             0.000000
loser_rank            10.447925
loser_rank_points     10.447925
loser_seed            78.698938
match_num              0.000000
round                  0.000000
score                  0.000000
surface                0.000000
tourney_date           0.000000
tourney_id             0.000000
tourney_level          0.000000
tourney_name           0.000000
winner_age             0.054759
winner_entry          88.062644
winner_hand            1.686562
winner_ht             61.362392
winner_id              0.000000
winner_ioc             0.000000
winner_name            0.000000
winner_rank            7.063848
winner_rank_points     7.063848
winner_seed           58.953017
dtype: float64

# 2015 - Little Odd!

In 2015, we have 49 attributes/fields. But some of them are completely empty (i.e not even a single datapoint has values for those attributes). Since there are 2651 empty values in thie year, it does affect the overall sparsity values given for 2003-16

In [9]:
df = pd.read_csv('wta_matches_2015.csv')
df.shape

(2651, 49)

In [10]:
df.isna().mean()*100

tourney_id              0.000000
tourney_name            0.000000
surface                 0.000000
draw_size               0.000000
tourney_level           0.000000
tourney_date            0.000000
match_num               0.000000
winner_id               0.000000
winner_seed            56.356092
winner_entry           85.665786
winner_name             0.000000
winner_hand             0.188608
winner_ht              32.780083
winner_ioc              0.000000
winner_age              0.000000
winner_rank             0.452659
winner_rank_points      0.452659
loser_id                0.000000
loser_seed             75.292343
loser_entry            77.291588
loser_name              0.000000
loser_hand              0.452659
loser_ht               43.304413
loser_ioc               0.000000
loser_age               0.000000
loser_rank              1.018484
loser_rank_points       1.018484
score                   0.000000
best_of                 0.000000
round                   0.000000
minutes   

# Sparsity Statistics from 2003 to 2014 and 2016

In [11]:
df2003 = pd.read_csv('wta_matches_2003.csv')
df2004 = pd.read_csv('wta_matches_2004.csv')
df2005 = pd.read_csv('wta_matches_2005.csv')
df2006 = pd.read_csv('wta_matches_2006.csv')
df2007 = pd.read_csv('wta_matches_2007.csv')
df2008 = pd.read_csv('wta_matches_2008.csv')
df2009 = pd.read_csv('wta_matches_2009.csv')
df2010 = pd.read_csv('wta_matches_2010.csv')
df2011 = pd.read_csv('wta_matches_2011.csv')
df2012 = pd.read_csv('wta_matches_2012.csv', encoding = "ISO-8859-1")
df2013 = pd.read_csv('wta_matches_2013.csv')
df2014 = pd.read_csv('wta_matches_2014.csv')
df2016 = pd.read_csv('wta_matches_2016.csv')

frames = [df2003, df2004, df2005, df2006, df2007, df2008, df2009, df2010, df2011, df2012, df2013, df2014, df2016] 
df = pd.concat(frames, sort = 'False')
df.shape

(33805, 49)

In [12]:
df.isna().mean()*100

best_of                0.000000
draw_size              0.000000
l_1stIn               41.523443
l_1stWon              41.523443
l_2ndWon              41.523443
l_SvGms               99.695311
l_ace                 41.544150
l_bpFaced             41.529360
l_bpSaved             41.529360
l_df                  41.629936
l_svpt                41.526401
loser_age              0.011833
loser_entry           79.523739
loser_hand             1.831090
loser_ht              38.281319
loser_id               0.000000
loser_ioc              0.000000
loser_name             0.000000
loser_rank             4.898684
loser_rank_points      4.898684
loser_seed            77.213430
match_num              0.000000
minutes               99.695311
round                  0.000000
score                  0.000000
surface                0.127200
tourney_date           0.000000
tourney_id             0.000000
tourney_level          0.000000
tourney_name           0.000000
w_1stIn               41.523443
w_1stWon

# Other Statistics from 2003 to 2014 and 2016

In [13]:
df['tourney_name'].value_counts()

Wimbledon                     1524
Australian Open               1524
French Open                   1524
US Open                       1524
Miami                         1140
Indian Wells                  1140
Birmingham                     660
Charleston                     660
Canadian Open                  652
Beijing                        489
Cincinnati                     482
Tokyo                          464
Auckland                       403
Madrid                         399
Rome                           385
Doha                           384
Los Angeles                    377
Hobart                         372
Tashkent                       372
Bogota                         368
Quebec City                    368
Acapulco                       368
Strasbourg                     360
Luxembourg                     356
Linz                           348
Eastbourne                     348
Pattaya                        341
Palermo                        341
Seoul               

In [14]:
df['tourney_level'].value_counts()

G     6096
I     5883
P     3878
D     3808
T1    3194
T3    3119
T2    2824
T4    1982
PM    1864
T5     448
CC     216
O      189
W      180
CH     124
Name: tourney_level, dtype: int64

In [15]:
df['winner_name'].value_counts()

Maria Sharapova                 519
Jelena Jankovic                 506
Serena Williams                 475
Svetlana Kuznetsova             448
Caroline Wozniacki              418
Flavia Pennetta                 412
Vera Zvonareva                  407
Agnieszka Radwanska             406
Nadia Petrova                   402
Marion Bartoli                  402
Ana Ivanovic                    393
Daniela Hantuchova              391
Francesca Schiavone             380
Elena Dementieva                377
Venus Williams                  367
Samantha Stosur                 366
Victoria Azarenka               365
Na Li                           350
Kim Clijsters                   332
Justine Henin                   321
Patty Schnyder                  314
Amelie Mauresmo                 312
Anabel Medina Garrigues         295
Shahar Peer                     294
Dinara Safina                   292
Maria Kirilenko                 288
Sara Errani                     275
Lucie Safarova              

In [16]:
df['loser_name'].value_counts()

Daniela Hantuchova               298
Francesca Schiavone              291
Jelena Jankovic                  279
Anabel Medina Garrigues          264
Marion Bartoli                   263
Klara Koukalova                  258
Flavia Pennetta                  255
Samantha Stosur                  236
Nadia Petrova                    233
Maria Kirilenko                  229
Svetlana Kuznetsova              222
Iveta Benesova                   222
Jill Craybas                     221
Shahar Peer                      217
Roberta Vinci                    216
Lucie Safarova                   214
Shuai Peng                       206
Patty Schnyder                   201
Vera Zvonareva                   200
Jie Zheng                        200
Dominika Cibulkova               190
Elena Vesnina                    190
Ana Ivanovic                     189
Tsvetana Pironkova               188
Sara Errani                      187
Virginie Razzano                 187
Gisela Dulko                     183
E

In [17]:
all_match_players = pd.concat([df['winner_name'], df['loser_name']])
print(all_match_players.shape)  
print(df.shape[0] * 2)

(67610,)
67610


In [18]:
#list of players
players = []
players = all_match_players.unique()
print(len(players))

1591


In [19]:
print("Player \t Number of matches in 2003-2014;2016")
player_match_count = all_match_players.value_counts()
print(player_match_count)

Player 	 Number of matches in 2003-2014;2016
Jelena Jankovic                785
Daniela Hantuchova             689
Francesca Schiavone            671
Svetlana Kuznetsova            670
Flavia Pennetta                667
Marion Bartoli                 665
Maria Sharapova                654
Nadia Petrova                  635
Vera Zvonareva                 607
Samantha Stosur                602
Caroline Wozniacki             586
Ana Ivanovic                   582
Agnieszka Radwanska            580
Serena Williams                563
Anabel Medina Garrigues        559
Elena Dementieva               542
Maria Kirilenko                517
Patty Schnyder                 515
Klara Koukalova                515
Shahar Peer                    511
Victoria Azarenka              506
Na Li                          505
Lucie Safarova                 489
Venus Williams                 487
Roberta Vinci                  467
Sara Errani                    462
Shuai Peng                     453
Dinara Saf

In [20]:
print("NUMBER OF PLAYERS WITH MORE THAN ONE MATCHES \n \n")
for i in range(1591):
    if player_match_count[players[i]] > 1:
        print('{:<30s} {:<10d}'.format(players[i], player_match_count[players[i]])) 
        

NUMBER OF PLAYERS WITH MORE THAN ONE MATCHES 
 

Serena Williams                563       
Marie Gaiane Mikaelian         59        
Barbara Schett                 79        
Clarisa Fernandez              51        
Meghann Shaughnessy            246       
Ludmila Cervanova              95        
Sandra Kleinova                31        
Ai Sugiyama                    364       
Daniela Hantuchova             689       
Ashley Harkleroad              122       
Magui Serna                    98        
Anna Smashnova                 201       
Emilie Loit                    245       
Fabiola Zuluaga                116       
Lina Krasnoroutskaya           63        
Amelie Mauresmo                421       
Justine Henin                  366       
Jelena Kostanic Tosic          204       
Stephanie Foretz Gacon         136       
Dally Randriantefy             69        
Patty Schnyder                 515       
Stephanie Cohen Aloro          120       
Tathiana Garbin            